confgenerator/filter/internal/ast/ast.go

```diff
@@ -0,0 +1,435 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ast
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/filter/internal/generated/token"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
+)
+
+type Attrib interface{}
+
+type Target []string
+
+var logEntryRootValueMapToFluentBit = map[string]string{
+	"severity": "logging.googleapis.com/severity",
+}
+
+var logEntryRootStructMapToFluentBit = map[string]string{
+	"labels":         "logging.googleapis.com/labels",
+	"operation":      "logging.googleapis.com/operation",
+	"sourceLocation": "logging.googleapis.com/sourceLocation",
+	"httpRequest":    "logging.googleapis.com/http_request",
+}
+
+// RecordAccessor returns a string that can be used as a key in fluentbit config
+func (m Target) RecordAccessor() string {
```

Comment:
`RecordAccessor` and `logEntryToFluentBit` seem to always be used in combination; can you just make `RecordAccessor` call `logEntryToFluentBit` and thereby get rid of the two possible states a `Target` can be in? (I think `logEntryToFluentBit` could then return `([]string, error)`.)

Reply:
I'll do this, and I can even go a step further by eliminating `logEntryToFluentBit` entirely and changing `RecordAccessor` to return `(string, error)`. The intermediate `[]string` representation isn't useful anywhere (but I do still want the error returned for validation).